designer/server/src/main/scala/pl/touk/nussknacker/ui/api/NodesApiEndpoints.scala

```diff
@@ -0,0 +1,497 @@
+          validationPerformed = node.validationPerformed
+        )
+      }
+
+    }
+
+    // Only used in response, no need for valid decoder
+    @derive(encoder, schema)
+    final case class UIParameterDto(
+        name: String,
+        typ: TypingResult,
+        editor: ParameterEditor,
+        defaultValue: Expression,
+        additionalVariables: Map[String, TypingResult],
+        variablesToHide: Set[String],
+        branchParam: Boolean,
+        hintText: Option[String],
+        label: String
+    )
+
+    object UIParameterDto {
+      implicit lazy val parameterEditorSchema: Schema[ParameterEditor]    = Schema.derived
+      implicit lazy val dualEditorSchema: Schema[DualEditorMode]          = Schema.string
+      implicit lazy val expressionSchema: Schema[Expression]              = Schema.derived
+      implicit lazy val timeSchema: Schema[java.time.temporal.ChronoUnit] = Schema.anyObject
+
+      def apply(param: UIParameter): UIParameterDto = new UIParameterDto(
+        param.name,
+        param.typ,
+        param.editor,
+        param.defaultValue,
+        param.additionalVariables,
+        param.variablesToHide,
+        param.branchParam,
+        param.hintText,
+        param.label
+      )
+
+    }
+
+    @derive(schema, encoder, decoder)
+    final case class PropertiesValidationRequestDto(
+        additionalFields: ProcessAdditionalFields,
+        name: ProcessName
+    )
+
+    // Request doesn't need valid encoder
+    @derive(schema, decoder)
+    final case class ParametersValidationRequestDto(
+        parameters: List[UIValueParameterDto],
+        variableTypes: Map[String, TypingResultInJson]
+    )
+
+    implicit val parametersValidationRequestDtoEncoder: Encoder[ParametersValidationRequestDto] =
+      Encoder.encodeJson.contramap[ParametersValidationRequestDto](_ => Json.Null)
+
+    @derive(schema, encoder, decoder)
+    final case class ParametersValidationResultDto(
+        validationErrors: List[NodeValidationError],
+        validationPerformed: Boolean
+    )
+
+    // Request doesn't need valid encoder
+    @derive(schema, decoder)
+    final case class UIValueParameterDto(
+        name: String,
+        typ: TypingResultInJson,
+        expression: Expression
+    )
+
+    implicit lazy val expressionSchema: Schema[Expression]           = Schema.derived
+    implicit lazy val caretPosition2dSchema: Schema[CaretPosition2d] = Schema.derived
+
+    // Request doesn't need valid encoder
+    @derive(schema, decoder)
+    final case class ExpressionSuggestionRequestDto(
+        expression: Expression,
+        caretPosition2d: CaretPosition2d,
+        variableTypes: Map[String, TypingResultInJson]
+    ) {
+
+      def decodedVariableTypes(decoder: Decoder[TypingResult]): Map[String, TypingResult] =
+        variableTypes.map { case (key, result) =>
+          (key, decoder.decodeJson(result).getOrElse(Unknown))
+        } // todo unknown for now
```

Comment:
Is it ok to Uknown TypingResult to stay? Or should we throw some exception instead?